arg: fix order to use short form before long form

ServeurpersoCom opened the pull request:
*Make sure to read the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md) before submitting a PR*

Minor cosmetics for logs :
Fixes argument order to follow convention: short form before long form (e.g., `{"-x", "--xxx"}` instead of `{"--xxx", "-x"}`).

This affects how arguments appear in logs since `to_string()` uses `.back()`.

Found via: `grep -n '{"--[a-z-]*", "-[a-z]' common/arg.cpp`

Fixed 8 arguments:
- `-cram, --cache-ram`
- `-kvu, --kv-unified`
- `-ot, --override-tensor`
- `-otd, --override-tensor-draft`
- `-cmoe, --cpu-moe`
- `-ncmoe, --n-cpu-moe`
- `-cmoed, --cpu-moe-draft`
- `-ncmoed, --n-cpu-moe-draft`

ServeurpersoCom committed:
arg: fix order to use short form before long form

ServeurpersoCom committed:
arg: update doc

ngxson commented:
I think it's worth adding a check in `test-arg-parser` too:
- If the option has more than one argument, make sure the first argument is shorter than the second
- Apply the same check with `arg_neg`

ServeurpersoCom commented:
"the SRV_INF macro should already be prefixed with the function name, so I think this is not necessary:
#define SRV_INF(fmt, ...) LOG_INF("srv  %12.*s: " fmt, 12, __func__, __VA_ARGS__)"
Nice!

This PR just need test-arg-parser order with arg_neg

ServeurpersoCom committed:
arg: update test-arg-parser

ngxson commented on tests/test-arg-parser.cpp:
```diff
@@ -37,6 +37,38 @@ int main(void) {
+
+                    // only check if we have a mix of short (-x) and long (--xxx) forms
+                    bool first_is_short = (first.size() >= 2 && first[0] == '-' && first[1] != '-');
+                    bool last_is_short = (last.size() >= 2 && last[0] == '-' && last[1] != '-');
```
hmm, why do we need to allow last arg to be short? if `opt.args.size() > 1` then it's guaranteed to have first=short and last=long

I think just check if `first.length() > last.length()` is enough

ServeurpersoCom replied:
```
(root|~/llama.cpp.pascal) grep -E '"-[^"]*", "-[^"]*", "-[^"]*"' common/arg.cpp | while read line; do
    args=($(echo "$line" | grep -oP '"-[^"]*"' | tr -d '"'))
    printf "%s: " "$line"
    for arg in "${args[@]}"; do printf "%d " "${#arg}"; done
    echo ""
done
{"-h", "--help", "--usage"},: 2 6 7
{"-n", "--predict", "--n-predict"}, "N",: 2 9 11
{"-ngl", "--gpu-layers", "--n-gpu-layers"}, "N",: 4 12 14
{"-hf", "-hfr", "--hf-repo"}, "<user>/<model>[:quant]",: 3 4 9
{"-hfd", "-hfrd", "--hf-repo-draft"}, "<user>/<model>[:quant]",: 4 5 15
{"-hfv", "-hfrv", "--hf-repo-v"}, "<user>/<model>[:quant]",: 4 5 11
{"-o", "--output", "--output-file"}, "FNAME",: 2 8 13
{"-v", "--verbose", "--log-verbose"},: 2 9 13
{"-lv", "--verbosity", "--log-verbosity"}, "N",: 3 11 15
{"--draft", "--draft-n", "--draft-max"}, "N",: 7 9 11
{"-ngld", "--gpu-layers-draft", "--n-gpu-layers-draft"}, "N",: 5 18 20
(root|~/llama.cpp.pascal)
```
Want to check for good sorting ?
🤣

ngxson replied:
I don't mean we need to sort in term of increase in length, just first < last is enough, we don't care about all the args in-between

ofc there are some exceptions as you pointed out where first == last:
- --ctx-checkpoints vs --swa-checkpoints
- --image vs --audio

Which I think can simply be ok to lossen the condition to first <= last

some args like --image, --audio doesn't have short form, so it's fine to **skip** the check if the first arg is short (e.g. no need to check if arg start with only one dash)

ServeurpersoCom committed:
arg: address review feedback from ngxson

simplified to check first.length() <= last.length() only
fixed: --sampler-seq, --rerank, --draft ordering
note: middle positions in 3+ arg sets are not verified

ServeurpersoCom committed:
arg: update doc
